src/app/shared/form/builder/ds-dynamic-form-ui/models/autocomplete/dynamic-autocomplete.component.ts

```diff
@@ -0,0 +1,296 @@
+import {PaginationComponentOptions} from '../../../../../pagination/pagination-component-options.model';
+import {AUTOCOMPLETE_COMPLEX_PREFIX} from './dynamic-autocomplete.model';
+import {EU_PROJECT_PREFIX, SEPARATOR, SPONSOR_METADATA_NAME} from '../ds-dynamic-complex.model';
+import {VocabularyEntry} from '../../../../../../core/submission/vocabularies/models/vocabulary-entry.model';
+import {DynamicAutocompleteService} from './dynamic-autocomplete.service';
+
+/**
+ * Component representing a autocomplete input field
+ */
+@Component({
+  selector: 'ds-dynamic-autocomplete',
+  styleUrls: ['../tag/dynamic-tag.component.scss'],
+  templateUrl: './dynamic-autocomplete.component.html'
+})
+export class DsDynamicAutocompleteComponent extends DsDynamicTagComponent implements OnInit {
+
+  @Input() bindId = true;
+  @Input() group: FormGroup;
+  @Input() model: DynamicTagModel;
+
+  @Output() blur: EventEmitter<any> = new EventEmitter<any>();
+  @Output() change: EventEmitter<any> = new EventEmitter<any>();
+  @Output() focus: EventEmitter<any> = new EventEmitter<any>();
+
+  @ViewChild('instance') instance: NgbTypeahead;
+
+  hasAuthority: boolean;
+  isSponsorInputType = false;
+
+  searching = false;
+  searchFailed = false;
+  currentValue: any;
+  public pageInfo: PageInfo;
+
+  constructor(protected vocabularyService: VocabularyService,
+              protected cdr: ChangeDetectorRef,
+              protected layoutService: DynamicFormLayoutService,
+              protected validationService: DynamicFormValidationService,
+              protected metadataValueService: MetadataValueDataService,
+              private lookupRelationService: LookupRelationService
+  ) {
+    super(vocabularyService, cdr, layoutService, validationService);
+  }
+
+  /**
+   * Initialize the component, setting up the init form value
+   */
+  ngOnInit(): void {
+    if (isNotEmpty(this.model.value)) {
+      if (this.model.value instanceof FormFieldMetadataValueObject && isNotEmpty(this.model.value.value)) {
+        this.model.value = this.model.value.value;
+      }
+      this.setCurrentValue(this.model.value, true);
+    }
+    if (isNotEmpty(this.model.metadataFields) && isNotEmpty(this.model.metadataFields[0])) {
+      this.isSponsorInputType = this.model.metadataFields[0] === SPONSOR_METADATA_NAME;
+    }
+  }
+
+  /**
+   * Updates model value with the selected value
+   * @param event The value to set.
+   */
+  onSelectItem(event: NgbTypeaheadSelectItemEvent) {
+    this.updateModel(event.item);
+    this.cdr.detectChanges();
+  }
+
+  onBlur(event: Event) {
+    this.dispatchUpdate(this.currentValue);
+    this.cdr.detectChanges();
+  }
+
+  updateModel(updateValue) {
+    let newValue = updateValue.display;
+    if (this.isSponsorInputType) {
+      if (updateValue instanceof VocabularyEntry) {
+        newValue = AUTOCOMPLETE_COMPLEX_PREFIX + SEPARATOR + updateValue.value;
+      } else {
+        // special autocomplete sponsor input
```

Comment:
if it is special, this should be guarded by if (...)

Reply:
refactored - special entry is ExternalSourceEntry. I added a check for that.